src/v2/providers/tasks.ts

```diff
@@ -251,8 +251,21 @@ export function onTaskDispatched<Args = any>(
     taskQueueTrigger: initTaskQueueTrigger(options.getGlobalOptions(), opts),
   };
 
-  copyIfPresent(func.__endpoint.taskQueueTrigger, opts, "retryConfig");
-  copyIfPresent(func.__endpoint.taskQueueTrigger, opts, "rateLimits");
+  copyIfPresent(
+    func.__endpoint.taskQueueTrigger.retryConfig,
+    opts.retryConfig || {},
```

Comment:
q: why do we need `|| {}`? If `retryConfig` doesn't exist, then `copyIfPresent` would be a no-op right?

Reply:
Ahh thanks for the catch, fixed!